spec.html

```diff
@@ -3784,7 +3780,30 @@ <h1>Runtime Semantics: MV</h1>
+          <p>For the MV of any production not shown above, see <emu-xref href="#sec-static-semantics-mv"></emu-xref>.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-stringtonumber" aoid="StringToNumber">
+          <h1>StringToNumber ( _str_ )</h1>
+          <emu-alg>
+            1. Assert: Type(_str_) is String.
+            1. Let _text_ be the sequence of Unicode code points that results from interpreting _str_ as UTF-16 encoded Unicode text as described in <emu-xref href="#sec-ecmascript-language-types-string-type"></emu-xref>.
+            1. Let _literal_ be the result of parsing _text_ using the goal symbol |StringNumericLiteral|. If _text_ does not conform to the grammar, or if any elements of _text_ were not matched by the parse, return *NaN*.
+            1. NOTE: The terminal symbols of the |StringNumericLiteral| grammar are all code points in the Unicode Basic Multilingual Plane (BMP). Therefore, this operation will return *NaN* if _str_ contains any <emu-xref href="#leading-surrogate"></emu-xref> or <emu-xref href="#trailing-surrogate"></emu-xref> code units, whether paired or unpaired.
+            1. If _literal_ contains a |StrUnsignedDecimalLiteral| and _literal_ has more than 20 significant digits, then
+              1. Let _lit_ be an implementation-dependent choice of:
+                * _literal_
+                * a literal produced from _literal_ by replacing each significant digit after the 20th with a `0` digit
+                * a literal produced from _literal_ by replacing each significant digit after the 20th with a `0` digit and then incrementing the literal at the 20th significant digit position
+            1. Else,
+              1. Let _lit_ be _literal_.
+            1. Let _mv_ be the MV of _lit_.
+            1. If _mv_ is 0, then
+              1. If the first non white space code point in _text_ is `"-"`, return *-0*; else return *+0*.
+            1. Return the Number value for _mv_ (as specified in <emu-xref href="#sec-ecmascript-language-types-number-type"></emu-xref>).
+          </emu-alg>
+
+          <p>A digit is significant if it is not part of an |ExponentPart| and</p>
```

Comment:
This could benefit from either being converted into a linkable definition or moving above the algorithm steps. It feels very similar to "the string-concatenation of…".

Reply:
By "this", do you mean just the definition of significant digit?

Reply:
What I mean is that I think the overall state of spec with this PR would benefit from either making a single linkable definition for "significant digit", or—failing that—continuing to define "significant digit" inside the operations that need it but moving that definition above the algorithm steps to establish context for when the concept is encountered inside them.

Reply:
I agree that it would be useful to have a single linkable definition for "significant digit". However, I think it actually could be somewhat tricky to write a definition that works for all uses, so I'd prefer to have that in a PR on its own, rather than complicating this PR.